_overviews/scala3-book/fp-pure-functions.md

```diff
@@ -50,24 +50,20 @@ Most methods on the Scala collections classes also work as pure functions, inclu
 
 Conversely, the following functions are _impure_ because they violate the definition.
 
-The `foreach` method on collections classes is impure because it’s only used for its side effects, such as printing to STDOUT.
+- `println` -- methods that interact with the console, files, databases, web services, sensors, etc., are all impure.
+- `getHour` -- date and time related methods are all impure because their output depends on something other than their input parameters 
```

Comment:
```suggestion
- `currentTimeMillis ` -- date and time related methods are all impure because their output depends on something other than their input parameters 
```